src/blockdata/transaction.rs

```diff
@@ -370,19 +415,24 @@ impl Transaction {
     /// - Does NOT attempt to support OP_CODESEPARATOR. In general this would require evaluating
     /// `script_pubkey` to determine which separators get evaluated and which don't, which we don't
     /// have the information to determine.
-    /// - Does NOT handle the sighash single bug, you should either handle that manually or use
-    /// [`Self::signature_hash()`] instead.
+    /// - Does NOT handle the sighash single bug (see "`Option` return value" section)
+    ///
+    /// # Return type
+    ///
+    /// This function can't handle the SIGHASH_SINGLE bug internally, so it returns [`EncodeSigningResult`]
```

Comment:
Should be `EncodeSigningDataResult`, also maybe recommend checking out `EncodeSigningDataResult::is_sighash_single_bug` for quick handling.